Standards/scs-0101-v1-entropy.md

```diff
@@ -5,114 +5,118 @@ status: Draft
+(available in Linux since kernel 5.4 or, before that, via a daemon such as
+[HavegeD](http://www.issihosts.com/haveged/)) as well as a special CPU
+instruction set (RDRAND/RDSEED), which is present in all modern CPUs from
+ARM, AMD, and Intel, and even in the consumer-grade Raspberry Pi
+(1b onwards).
+
+Finally, a dedicated device can be utilized — if present — that is
+called _hardware random number generator_ or HW RNG for short. For instance,
+the [trusted platform module](https://en.wikipedia.org/wiki/Trusted_Platform_Module)
+includes a HW RNG. On Linux systems, the HW RNG appears as `/dev/hwrng`.
+Note that, while the RDRAND/RDSEED instruction set can be construed as
+a HW RNG, it is not treated as such by the kernel, i.e., it _does not_
+appear as `/dev/hwrng`!
+
+The Linux kernel combines multiple sources of entropy into a pool. To this
+end, it will use all of the sources discussed so far with one exception:
+the HW RNG must be fed into the pool (if so desired) via the daemon `rngd`.
+The kernel converts the entropy from the pool into cryptographically
+secure random numbers that appear under `/dev/random`.
+
+With kernel 5.18, the algorithm that accomplishes
+said conversion has been drastically improved (see
+[linux-rng-5.17-18](https://web.archive.org/web/20230321040526/https://www.zx2c4.com/projects/linux-rng-5.17-5.18/)),
+so much so that running out of entropy is virtually ruled out.
+These patches have now also arrived in the upstream LTS images.
+
+### Entropy in virtual instances
+
+Virtual instances or virtual machines do not have the traditional sources
+of entropy mentioned above. However, the more recent methods mentioned
+above do work just fine (the RDRAND instruction set is not privileged).
+
+Alternatively, a virtualized HW RNG called `virtio-rng` can be established
+that injects entropy from the host into the instance, where this
+entropy can be sourced optionally from either the host's `/dev/random` or
+some HW RNG in the host. This virtualized HW RNG behaves just like real
+one, that is, it appears as `/dev/hwrng`, and the daemon `rngd` must
+be used to feed it into the kernel's entropy pool.
+
+In summary, with current kernels and CPUs entropy in virtual instances
+is readily available to a sufficient degree. In addition, the host's
+entropy sources can be injected using `virtio-rng` if so desired, e.g.,
+to enable access to a HW RNG.
+
+## Motivation
+
+As stated above, good sources of entropy are paramount for many
+important applications. Moreover, current technology makes it easy
+to provide these sources to virtual instances. Therefore, this standard
+mandates that these sources be made available on all conformant clouds.
+
+## Entropy in SCS clouds
+
+This standard does not intend to make any guarantees for linux images
+that are not themselves included in the
+[SCS standard](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0102-v1-image-metadata.md), in particular
```

Comment:
This one is indeed a bit tricky. I guess what I mean here is not the image metadata standard itself, but the **certification level _SCS compatible_**, which includes the image metadata standard. What the CSPs get certified for is the bundle as a whole. However, speaking of "SCS certification level _SCS compatible_" is a mouthful. Therefore I opted for SCS standard, but then only linked the relevant document.

Reply:
While working on https://github.com/SovereignCloudStack/issues/issues/343, I learned that there is such a thing as "the SCS standard", namely with respect to a given release number, and it comprises all standards that are denoted as stable in that release. Source: https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0001-v1-sovereign-cloud-standards.md#stabilized-phase-stable

So the wording "SCS standard" is not unprecedented and it is even well defined. The image metadata standard is stable, and I guess it will remain so for the upcoming release. Therefore it should be warranted to speak of the SCS standard at large while linking to a specific document that is part of the standard.